ingest/build-configs/ncbi/bin/parse-metadata-from-strain

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+"""
+Attempts to parse additional metadata from the `strain` field.
+"""
+import argparse
+import json
+from sys import stderr, stdin, stdout
+from typing import Optional, Dict
+
+
+EXPECTED_STRAIN_SPLIT_LENGTH = {4, 5}
+EXPECTED_STRAIN_SPLIT_PREFIX = "A"
+HUMAN_HOST = "Homo sapiens"
+NCBI_LOCATION_SEPARATOR = ":"
+
+
+def parse_strain(strain: str) -> Optional[Dict[str, str]]:
+    """
+    Parses the `STRAIN_METADATA` from the provided *strain*, with the expectation
+    that the strain name is formatted as:
+
+    For non-human hosts:
+
+        A / <host> / <location> / <sample id> / <year>
+
+    For human hosts:
+
+        A / <location> / <sample id> / <year>
+
+    """
+    split_strain = strain.split("/")
+
+    if (len(split_strain) not in EXPECTED_STRAIN_SPLIT_LENGTH
+            or split_strain[0] != EXPECTED_STRAIN_SPLIT_PREFIX):
+        print("WARNING: Unable to parse additional metadata from strain. ",
+              f"Encountered unexpected `strain` format {strain!r}",
+              file=stderr)
+        return None
+
+    # Assume human host format
+    if len(split_strain) == 4:
+        prefix, location, sample, year = split_strain
+        host = HUMAN_HOST
+    elif len(split_strain) == 5:
+        prefix, host, location, sample, year = split_strain
+    else:
+        raise ValueError(f"Encountered unsupported strain format {strain!r}")
+
+    return {
+        "host": host,
+        "location": location,
+        "date": year
+    }
+
+
+def update_record(record: Dict[str, str], strain_metadata: Dict[str, str]):
```

Comment:
```suggestion
def update_record(record: Dict[str, str], strain_metadata: Dict[str, str]) -> None:
```

Reply:
Included in https://github.com/nextstrain/avian-flu/pull/40/commits/eb5e068ef2d4260b01c7ac49fa805b539fb05e83